File: Python/Score Board/scoreBoard/ScoreBoard.py
class ScoreBoard:

    def __init__(self, capacity):
        self.total_entry = 0
        self.board = [None] * capacity


    def add(self, entry):
        new_score = entry.get_score()

        if(self.total_entry < len(self.board) or new_score > self.board[self.total_entry - 1].get_score()):
            if(self.total_entry < len(self.board)):
                self.total_entry+=1
            
            last_index = self.total_entry - 1

            while(last_index > 0 and new_score > self.board[last_index-1].get_score()):
                self.board[last_index] = self.board[last_index-1]
                last_index-=1
            self.board[last_index] = entry

    

    def remove(self, index):
        if(index < 0 or index >= self.total_entry):
            raise IndexError(f"Invalid index: {index}")
        
        removed_entry = self.board[index]

        for i in range(index, self.total_entry-1):
            self.board[i] = self.board[i+1]
        
        self.board[self.total_entry-1] = None
        self.total_entry-=1

    def get_entry(self, index):
        return self.board[index]
    
    def get_total_entry(self):
        return self.total_entry

File: Java/Score_Board/src/main/java/com/kucse/score_board/ScoreBoard.java
package com.kucse.score_board;


public class ScoreBoard {
    private int totalEntry = 0;
    private GameEntry[] board;

    public ScoreBoard(int capacity) {
        board = new GameEntry[capacity];
    }

    public void add(GameEntry e) {
        int newScore = e.getScore();

        if (totalEntry < board.length || newScore > board[totalEntry - 1].getScore()) {
            if (totalEntry < board.length) {
                totalEntry++;
            }
            int lastIndex = totalEntry - 1;

            while (lastIndex > 0 && board[lastIndex - 1].getScore() < newScore) {
                board[lastIndex] = board[lastIndex - 1];
                lastIndex--;
            }
            board[lastIndex] = e;
        }
    }

    public GameEntry remove(int index) throws IndexOutOfBoundsException {
        if (index < 0 || index >= totalEntry) {
            throw new IndexOutOfBoundsException("Invalid index: " + index);
        }
        GameEntry removedEntry = board[index];

        for (int i = index; i < totalEntry - 1; i++) {
            board[i] = board[i + 1];
        }
        board[totalEntry - 1] = null;
        totalEntry--;

        return removedEntry;
    }

    public int getTotalEntry() {
        return totalEntry;
    }

    public GameEntry getEntry(int index) {
        if (index >= 0 && index < totalEntry) {
            return board[index];
        }
        return null;
    }
}
